### ISSUE TITLE
Installation issue: parallelio

### ISSUE THREAD

**Issue by svahl991** [JCSDA/spack#17]
### Steps to reproduce the issue

```
$> spack install -v --fail-fast 2>&1 | tee log.install.pio
```
Representative error:
```
  >> 1832    /Users/vahl/spack_test/spack-stack/cache/build_stage/spack-stage-parallelio-2.5.4-yuoh3mtbqaadqfwhrwq25tpuyyc7ynbp/spack-src/src/clib/pio_get_vard.c:260:5
             : error: conflicting types for 'PIOc_get_vard'
     1833    int PIOc_get_vard(int ncid, int varid, int decompid, const PIO_Offset recnum,
     1834        ^
     1835    /Users/vahl/spack_test/spack-stack/cache/build_stage/spack-stage-parallelio-2.5.4-yuoh3mtbqaadqfwhrwq25tpuyyc7ynbp/spack-src/src/clib/pio.h:1189:9: note:
             previous declaration is here
     1836        int PIOc_get_vard(int ncid, int varid, int decompid, const PIO_Offset recnum, void *buf);
     1837            ^
     1838    12 errors generated.
```

### Information on your system

```
* **Spack:** 0.16.1-6556-853386bad6
* **Python:** 3.9.10
* **Platform:** darwin-catalina-skylake
* **Concretizer:** clingo
```

### Additional information

[install.pio.log](https://github.com/NOAA-EMC/spack/files/8207567/install.pio.log)
[packages.yaml.txt](https://github.com/NOAA-EMC/spack/files/8207584/packages.yaml.txt)
[spack-build-out.txt](https://github.com/NOAA-EMC/spack/files/8208755/spack-build-out.txt)
[spack-build-env.txt](https://github.com/NOAA-EMC/spack/files/8208762/spack-build-env.txt)

`spack maintainers parallelio` returns @tkameyama

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by kgerheiser** [comment]
Weird because they both have the same type signature.

Here's the definition in `/pio_get_vard.c:260:5`
```
int PIOc_get_vard(int ncid, int varid, int decompid, const PIO_Offset recnum,
                  void *buf)
{
    return PIOc_get_vard_tc(ncid, varid, decompid, recnum, NC_NAT, buf);
}
```

And here's the signature in the header `pio.h:1189:9`

```
int PIOc_get_vard(int ncid, int varid, int decompid, const PIO_Offset recnum, void *buf);
```

**Comment by kgerheiser** [comment]
Are you using mpich? Looks like it's this open issue in PIO https://github.com/NCAR/ParallelIO/issues/1912 where  MPI_Offset in `pio_internal.h` and `pio.h` are resolved to different types. 

If they have a fix for it we can create a patch.

**Comment by kgerheiser** [comment]
This should be fixed with #23
